src/components/tl/mlx5/mcast/p2p/ucc_tl_mlx5_mcast_p2p.c

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "ucc_tl_mlx5_mcast_p2p.h"
+
+static inline void ucc_tl_mlx5_mcast_ucx_completion_cb(void* context)
```

Comment:
`context` should be typed `ucc_tl_mlx5_mcast_p2p_completion_obj_t *`
The casting from `void*` could be done inside `ucc_tl_mlx5_mcast_completion_cb`

Reply:
The reason for void * as input is that we would like to have flexibility in the future. 
Imagine in the future, this p2p library implements 2 different functions as a completion function. Each of these completion functions casts input context to a different struct. Having void * as input enables us to have a single function declaration inside the p2p.h file. To better realize it, please take a look at libraries that take advantage of function pointers to realize several implementations for a single function.